### DISCUSSION TITLE
How to use MapQueryString with DTO containing Boolean with 'true' or 'false' value as querystring

### DISCUSSION THREAD

**Discussion by qvandekadsye** [symfony/symfony#52572] [Q&A]
Hi,

Considering this class

```php
public class MyDTO {
    public function __construct(public bool isFeatured) {}
    
}
```

If I use MapQueryString for the URL /mypage?isFeatured=0 or /mypage?isFeatured=1, the DTO is correctly hydrated. But not when the query string is `isFeatured=true` or `isFeatured=false` 

Is it expected? Is it a bug? What can I do to use `true` or `false` and the DTO hydrated.

Your sincerly

**Comment by yceruto** [comment]
The behavior you're experiencing is expected. PHP interprets query string parameters as strings `parse_str('isFeatured=false', $q)` and thus "true" and "false" are not automatically recognized as boolean values. 

When you use `MapQueryString` it attempts to map this string value (using the Serializer/PropertyAccess components) to the property specified in your DTO (of type `bool` in this case) and since PHP treats any non-empty string as `true` (except for the string `"0"` which is a special case and is considered `false`) in a boolean context `"true"` and `"false"` will both result in `true` when cast to boolean.

To handle this, you can implement custom logic in your DTO to manually convert the string `"false"` to their respective boolean value:
```php
class MyDTO 
{
    // ...

    public function setIsFeatured(string|bool $value): void
    {
        $this->isFeatured = is_string($value) ? 'false' !== $value && '0' !== $value : $value;
    }
}
```
If you need a more generic or global solution for the entire application, consider a custom `ObjectNormalizer`. However, be aware that this approach may introduce more complexity in implementing such an assumption, because inferring the property type is not part of the current logic and might decrease the denormalizing performance as well.

**Comment by ghost** [comment]
`public class`???
